layer/clients/model_catalog.py

```diff
@@ -134,29 +98,72 @@ def create_model_version(
         )
-        should_create_initial_train = model.training.hyperparameter_tuning is None
         response = self._service.CreateModelVersion(
             CreateModelVersionRequest(
                 model_path=model_path,
                 description=model.description,
-                training_files_hash=Sha256(value=model.training_files_digest),
-                should_create_initial_train=should_create_initial_train,
-                fabric=model.training.fabric,
+                training_files_hash=Sha256(value=model.source_code_digest.hexdigest()),
+                should_create_initial_train=True,
+                fabric=model.get_fabric(is_local=is_local),
             ),
         )
         self._logger.debug(f"CreateModelVersionResponse: {str(response)}")
         return response
 
-    def load_model_definition(self, path: str) -> ModelDefinition:
+    def store_training_metadata(
+        self,
+        model: ModelFunctionDefinition,
+        s3_path: S3Path,
+        version: ModelVersion,
+        is_local: bool,
+    ) -> None:
+        request: StoreTrainingMetadataRequest = StoreTrainingMetadataRequest(
+            model_version_id=version.id,
+            name=model.name,
+            description=model.description,
+            source_code_env=SourceCodeEnvironment(
+                source_code=SourceCode(
+                    remote_file_location=RemoteFileLocation(
+                        name=model.entrypoint,
+                        location=f"s3://{s3_path.bucket}/{s3_path.key}{model.name}.tgz",
+                    ),
+                    language=SourceCode.Language.Value("LANGUAGE_PYTHON"),
+                    language_version=SourceCode.LanguageVersion(
+                        major=int(model.language_version[0]),
+                        minor=int(model.language_version[1]),
+                        micro=int(model.language_version[2]),
+                    ),
+                ),
+                dependency_file=DependencyFile(
+                    name=model.environment,
+                    location=model.environment,
+                ),
+            ),
+            entrypoint=model.entrypoint,
+            fabric=model.get_fabric(is_local=is_local),
+        )
+        self._logger.debug(f"StoreTrainingMetadataRequest request: {str(request)}")
+        response = self._service.StoreTrainingMetadata(request)
+        self._logger.debug(f"StoreTrainingMetadata response: {str(response)}")
+
+    def load_model_by_path(self, path: str) -> Model:
         load_response = self._service.LoadModelTrainDataByPath(
             LoadModelTrainDataByPathRequest(path=path),
         )
-        return ModelDefinition(
-            name=path,
-            train_id=load_response.id,
-            proto_flavor=load_response.flavor,
-            s3_path=load_response.s3_path,
-            credentials=load_response.credentials,
+        flavor = get_flavor_for_proto(load_response.flavor)
```

Comment:
This function is only used here -- perhaps move it here and move the exception raise logic into it too?

Reply:
saving this for future cleanup